lib/bashcov/xtrace.rb

```diff
@@ -13,7 +15,10 @@ class Xtrace
     # @see http://stackoverflow.com/questions/59895/can-a-bash-script-tell-what-directory-its-stored-in
     # @note We use a forward slash as delimiter since it's the only forbidden
     #   character in filenames on Unix and Windows.
-    PS4 = %Q{#{PREFIX}${BASH_SOURCE[0]}/${LINENO}: }
+    _READLINK = find_executable('readlink')
+    _GREADLINK = find_executable('greadlink')
+    READLINK = _GREADLINK.nil? ? _READLINK : _GREADLINK
+    PS4 = %Q{#{PREFIX}$(#{READLINK} -f ${BASH_SOURCE[0]})/${LINENO}: }
```

Comment:
1. I'm not a big fan of including `mkmf` here - it's supposed to be used in Makefile scripts and the like but not in "normal" Ruby code. If you look [here](https://github.com/ruby/ruby/blob/trunk/lib/mkmf.rb), you can see it defines a lot of top-level stuff and monkey-patches core types like `String`.


1. Why would you use `greadlink`? Isn't `readlink` available on OS X?


1. Doesn't bash find the path to `readlink` if you simply do `PS4 = %Q{#{PREFIX}$(readlink -f ${BASH_SOURCE[0]})/${LINENO}: }`?


1. If not, does it work using something like `READLINK = IO.popen("whereis -b readlink").read.match(/\Areadlink:\s([^\s]+)/).captures.first`? I know it's kinda ugly and I'm sure there is a better way to do that but you get the idea.